modules/services/espanso.nix

```diff
@@ -16,9 +16,25 @@ let
     versionAtLeast
     ;
 
+  inherit (pkgs.stdenv.hostPlatform) isDarwin isLinux;
+
   cfg = config.services.espanso;
   espansoVersion = cfg.package.version;
 
+  package-bin =
+    if isLinux && cfg.x11Support && cfg.waylandSupport then
+      pkgs.writeShellScriptBin "espanso" ''
+        if [ -n "$WAYLAND_DISPLAY" ]; then
+          ${lib.meta.getExe cfg.package-wayland} "$@"
+        else
+          ${lib.meta.getExe cfg.package} "$@"
+        fi
+      ''
+    else if isLinux && cfg.waylandSupport then
+      cfg.package-wayland
```

Comment:
Missed this previously. will not allow user to override `package-wayland` but works because the default is still correct.